public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -907,30 +908,39 @@ public SurveyQuestionResult apply(AchillesResult o) {
         return ResponseEntity.ok(TO_CLIENT_ACHILLES_RESULT.apply(result));
     }
 
-    public TreeSet<Float> makeBins(Float min,Float max) {
-        TreeSet<Float> bins = new TreeSet<>();
+    public ArrayList<Float> makeBins(Float min,Float max) {
+        TreeSet<String> bins = new TreeSet<>();
         float binWidth = (max-min)/11;
 
-        if (min >= 10 && max >= 10) {
-            double tempBinWidth = Math.ceil(binWidth/10)*10;
-            binWidth = (float)tempBinWidth;
-        } else if ((max-min) <= 1 && (max-min) >= 0.1) {
-            double tempBinWidth = Math.ceil(binWidth/0.1)*0.1;
-            binWidth = (float)tempBinWidth;
+        if (binWidth >= 5) {
+            binWidth = Math.round(binWidth/5)*5;
+        } else if (binWidth >= 0.1 && binWidth <= 1) {
+            binWidth = Float.parseFloat(String.valueOf(Math.round(binWidth * 10.0) / 10.0));
```

Comment:
This may be what you meant in the above comment 

> Not dealing with rest of them right now, so just leaving them as they are.

 but the `else if` can be removed since it executes the same code as the `else`